src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -296,12 +299,24 @@ internal ValueTask<FlushResult> FlushAsync(CancellationToken cancellationToken)
             ValueTask<FlushResult> result;
             lock (_sync)
             {
-                CommitUnsynchronized();
+                var bytesWritten = CommitUnsynchronized();
```

Comment:
Nit: bytesWritten seems like it should be a number. Maybe invert the logic and call it wasEmpty.